src/views/Dashboard/GeneticAnalyst/Services/index.vue

```diff
@@ -198,171 +218,219 @@ export default {
+        this.wallet,
+        this.serviceId
+      )
+      this.txWeight = `${Number(
+        this.web3.utils.fromWei(String(txWeight.partialFee), "ether")
+      ).toFixed(4)} DBIO`
     },
 
     async getServiceList() {
       this.items = []
-      const analystDetail = await queryGeneticAnalystByAccountId(this.api, this.wallet.address)
+      const analystDetail = await queryGeneticAnalystByAccountId(
+        this.api,
+        this.wallet.address
+      )
       this.serviceList = analystDetail.services
 
       for (const serviceId of this.serviceList) {
-        const serviceDetail = await queryGeneticAnalystServicesByHashId(this.api, serviceId)
+        const serviceDetail = await queryGeneticAnalystServicesByHashId(
+          this.api,
+          serviceId
+        )
         const {
           id,
-          info: { description, name: serviceName, testResultSample }
+          info: {description, name: serviceName, testResultSample}
         } = serviceDetail
 
-        const price = this.formatPrice(serviceDetail.info.pricesByCurrency[0].totalPrice)
+        const price = this.formatPrice(
+          serviceDetail.info.pricesByCurrency[0].totalPrice
+        )
         const currency = serviceDetail.info.pricesByCurrency[0].currency
         const duration = `${serviceDetail.info.expectedDuration.duration} ${serviceDetail.info.expectedDuration.durationType}`
-        
-        const service = { id, description, serviceName, testResultSample, price, duration, currency }
+
+        const service = {
+          id,
+          description,
+          serviceName,
+          testResultSample,
+          price,
+          duration,
+          currency
+        }
         this.items.push(service)
       }
     },
 
+    async getGAOrders() {
+      const orders = await GAGetOrders()
+
+      this.orders = orders
+    },
+
     formatPrice(price) {
       return this.web3.utils.fromWei(String(price.replaceAll(",", "")), "ether")
     },
 
     async showDialog(id) {
-      this.serviceId = id  
+      this.serviceId = id
       this.showModal = true
       await this.getDeleteServiceFee()
     },
 
-    async onDelete() {
-      try{
+    onDelete() {
+      const isActiveOrder = this.orders?.data?.some(
+        (order) =>
+          order._source.service_id === this.serviceId &&
+          order._source.status === "Paid"
+      )
+      
+      if (isActiveOrder) {
+        this.showActiveOrder = true
+      } else {
+        this.deleteService()
+      }
+    },
+
+    async deleteService() {
+      try {
         this.isLoading = true
         await deleteGeneticAnalystService(this.api, this.wallet, this.serviceId)
       } catch (e) {
-        this.error = await(errorHandler(e.message))
+        this.error = await errorHandler(e.message)
         this.showModal = false
         this.isLoading = false
       }
+    },
+
+    goToDashboard() {
+      this.$router.push({name: "dashboard"})
+      this.showActiveOrder = false
```

Comment:
What do you mean by this function to throw to the destination route (dashboard)? just a suggestion, but you also need to double-check, I don't think you need to set `showActiveOrder` again because it will reset to `false` after you leave this page as long you set this default data to `false`

Reply:
updated